clash-cores/src/Clash/Cores/UART.hs

```diff
@@ -0,0 +1,211 @@
+{-|
+  Copyright   :  (C) 2021, LUMI GUIDE FIETSDETECTIE B.V.
+  License     :  BSD2 (see the file LICENSE)
+  Maintainer  :  Rowan Goemans <goemansrowan@gmail.com>
+
+  UART transmitter and receiver module
+-}
+
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE AllowAmbiguousTypes #-}
```

Comment:
I'm guessing this is for `baudRate`? Could we make that an `SNat`? Ambiguous types can make debugging type errors incredibly hard/unexpected.

Reply:
Yes! It works without it if `SNat` is used. So fixed.